WordPressKit/WordPressComServiceRemote+SiteVerticals.swift

```diff
@@ -0,0 +1,155 @@
+import Foundation
+
+import WordPressShared
+
+// MARK: - SiteVerticalsRequest
+
+/// Allows the construction of a request for site verticals.
+///
+/// NB: The default limit (5) applies to the number of results returned by the service. If a search with limit n evinces no exact match, (n - 1) server-unique results are returned.
+///
+public struct SiteVerticalsRequest: Encodable {
+    public let search: String
+    public let limit: Int
+
+    public init(search: String, limit: Int = 5) {
+        self.search = search
+        self.limit = limit
+    }
+}
+
+// MARK: - SiteVertical(s) : Response
+
+/// Models a Site Vertical
+///
+public struct SiteVertical: Decodable, Equatable {
+    public let identifier: String   // vertical IDs mix parent/child taxonomy (String)
+    public let title: String
+    public let isNew: Bool
+
+    public init(identifier: String,
+                title: String,
+                isNew: Bool) {
+
+        self.identifier = identifier
+        self.title = title
+        self.isNew = isNew
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case identifier = "vertical_id"
+        case title      = "vertical_name"
+        case isNew      = "is_user_input_vertical"
+    }
+}
+
+// MARK: - WordPressComServiceRemote (Site Verticals)
+
+/// Describes the errors that could arise when searching for site verticals.
+///
+/// - requestEncodingFailure:   unable to encode the request parameters.
+/// - responseDecodingFailure:  unable to decode the server response.
+/// - serviceFailure:           the service returned an unexpected error.
+///
+public enum SiteVerticalsError: Error {
+    case requestEncodingFailure
+    case responseDecodingFailure
+    case serviceFailure
+}
+
+/// Advises the caller of results related to requests for site verticals.
+///
+/// - success: the site verticals request succeeded with the accompanying result.
+/// - failure: the site verticals request failed due to the accompanying error.
+///
+public enum SiteVerticalsResult {
+    case success([SiteVertical])
+    case failure(SiteVerticalsError)
+}
+
+public typealias SiteVerticalsServiceCompletion = ((SiteVerticalsResult) -> ())
+
+/// Site verticals services, exclusive to WordPress.com.
+///
+public extension WordPressComServiceRemote {
+
+    /// Retrieves Verticals matching the specified criteria.
+    ///
+    /// - Parameters:
+    ///   - request:    the value object with which to compose the request.
+    ///   - completion: a closure including the result of the request for site verticals.
+    ///
+    func retrievalVerticals(request: SiteVerticalsRequest, completion: @escaping SiteVerticalsServiceCompletion) {
+
+        let endpoint = "verticals"
+        let path = self.path(forEndpoint: endpoint, withVersion: ._2_0)
```

Comment:
Oh this is great! At my last workplace we had a lot of issues due to the lack of API versioning and now I see the APi versions for ServiceRemoteWordPressComRESTApiVersion. This is great.